File: sources/js/test/Test_DoubleList.js
import DoubleList from "../list/DoubleList";

function TestDoubleList(){

    console.log("********** 顺序表测试开始 **********");

    let l = new DoubleList();

    console.log("追加元素-------------");
    l.append(17)
    l.append(20)
    l.display()

    console.log("插入元素-------------");
    l.insert(2, 18)
    l.insert(3, 19)
    l.display()

    console.log("删除元素-------------");
    l.delete(2)
    l.display()

    console.log("修改元素-------------");
    l.update(2, 22)
    l.display()

    console.log("查找元素前驱：", l.priorElem(22));
    console.log("查找元素后继：", l.nextElem(22));

    console.log("********** 顺序表测试结束 **********");
}

export default TestDoubleList

File: sources/go/test/Test_DoubleList.go
package test

import (
	"fmt"
	doublelist "structure/list/doublelist"
)

func TestDoubleList() {

	fmt.Printf("********** 测试开始 **********\n")

	l := doublelist.NewDoubleList()

	fmt.Println("追加元素-------------")
	l.Append(17)
	l.Append(20)
	l.Display()

	fmt.Println("插入元素-------------")
	l.Insert(2, 18)
	l.Insert(3, 19)
	l.Display()

	fmt.Println("删除元素-------------")
	l.Delete(1)
	l.Display()

	fmt.Println("修改元素-------------")
	l.Update(3, 22)
	l.Display()

	fmt.Println("按照位序查找元素-------------")
	ge, _ := l.GetElem(2)
	fmt.Printf("查找元素值：%d\n", ge)

	fmt.Println("按照值查找位序-------------")
	le, _ := l.LocateElem(18)
	fmt.Printf("查找元素位序：%d\n", le)

	fmt.Println("查找元素前驱-------------")
	pe, _ := l.PriorElem(24)
	fmt.Printf("查找元素前驱：%d\n", pe)

	fmt.Println("查找元素后继-------------")
	ne, _ := l.NextElem(18)
	fmt.Printf("查找元素前驱：%d\n", ne)

	fmt.Println("删除元素-------------")
	l.Delete(2)
	l.Display()

	fmt.Printf("********** 测试结束 **********\n")

}
